azavea/cac-tripplanner
# Feature/kak/show tours#1109

## Overview

 - Add tours to API results and homepage tab
 - Remove optional times from tour destinations, and add order and optional times to event destinations
 - Add detail page for tours
 - Modify detail page for events to list multiple destinations and optional times
 - Fix gulp watch
 - Upgrade to Django 2 (only needed to change string model methods)
 - Add tests for tours and event destinations


### Demo

Homepage tour tab:
![image](https://user-images.githubusercontent.com/960264/64642155-b0dbbc80-d3db-11e9-86b0-2014e58ada29.png)


Tour detail page lists locations in cards:

![image](https://user-images.githubusercontent.com/960264/64642272-f7311b80-d3db-11e9-8d64-6e9696a8b1a4.png)


Multi-day event:

![image](https://user-images.githubusercontent.com/960264/64642426-5a22b280-d3dc-11e9-98ab-2ea8c173e640.png)


Single day event:

![image](https://user-images.githubusercontent.com/960264/64642718-ff3d8b00-d3dc-11e9-959f-b0d9e242c25f.png)


### Notes

For now, directions to tours or events goes to the first location. Also, filtering by event or tour categories on the map page is now broken (shows no results), but works on the homepage. These can both be addressed in #1111 or #1121. 

There's no validation to check event destination times, if set, fall within range of the event times, or that they are ordered by time.


## Testing Instructions

 * Destroy and rebuild VM if not on Django 2 already, or else run migrations
 * In `app` VM: `cd /opt/app/src/ && npm run gulp` should build frontend and rebuild on JS or style file change
 * Page refresh after `gulp watch` rebuild should show change (there's no live reload)
 * Default event should still exist and have one destination set
 * Any other events that existed before migrations should still have their destinations set
 * Editing destinations on events and tours in the admin interface should work as expected
 * Create a tour and a multi-destination event
 * Up to two events, then up to two tours, should show at the top of the homepage results
 * Homepage after toggling to a filter tab then back to 'All' should look the same (JS vs Django template rendering)
 * Tour detail page should list destinations as cards in order
 * Multi-destination event page should list destinations, with times, if set
 * If there are more than four destinations for an event, there should be a 'See all' button with the right total count
 * Clicking the 'See all' button should hide itself and show the full destination list


## Checklist


Closes #1109 
Fixes #1036 
Closes #1110 
Closes #1138
